p2p/protocols/protocol.go

```diff
@@ -353,6 +353,9 @@ func (p *Peer) handleIncoming(handle func(ctx context.Context, msg interface{})
 	}
 
 	if p.spec.Hook != nil {
+		if wmsg.Size != uint32(len(wmsg.Payload)) {
+			log.Warn("Advertised message size and payload length don't match")
```

Comment:
this is not a warning its a major breach , should disconnect, punish, ...

Reply:
Ok. At this point why have a `Size` field at all then....we could always infer it by `len(wmsg,Payload)`